packages/yew/src/html/component/scope.rs

```diff
@@ -287,6 +287,76 @@ impl<COMP: BaseComponent> Scope<COMP> {
     pub fn send_message_batch(&self, messages: Vec<COMP::Message>) {
         self.arch_send_message_batch(messages)
     }
+
+    /// Creates a `Callback` which will send a message to the linked
+    /// component's update method when invoked, AND calls `.prevent_default()`
+    /// on the [web_sys::Event] that is passed to the callback.
+    ///
+    /// This is helpful for making more terse callback definitions for `onclick`
+    /// events on `<a>` elements, which alter the URL by default.
+    ///
+    /// If your app scrolls to the top of the page when a link or button with
+    /// a callback is clicked, this will prevent that.
+    ///
+    /// ### Short Example
+    /// ```rust,ignore
+    /// let onclick = ctx.link().callback_event(|e| RBMessage::Refresh);
+    ///
+    /// html! {
+    ///     <a href="#" {onclick}>{ "Refresh" }</a>
+    /// }
+    /// ```
+    /// ### Full Example
+    /// ```rust
+    /// # use yew::prelude::*;
+    /// # use web_sys::*;
+    /// pub struct RefreshButton;
+    ///
+    /// pub enum RBMessage {
+    ///     Refresh,
+    /// }
+    ///
+    /// impl Component for RefreshButton {
+    ///     type Message = RBMessage;
+    ///     type Properties = ();
+    ///
+    ///     fn create(ctx: &Context<Self>) -> Self {
+    ///         Self
+    ///     }
+    ///
+    ///     fn update(&mut self, ctx: &Context<Self>, msg: Self::Message) -> bool {
+    ///         match msg {
+    ///             RBMessage::Refresh => {
+    ///                 let win = window().unwrap();
+    ///                 let doc = win.document().unwrap();
+    ///                 let loc = doc.location().unwrap();
+    ///                 loc.reload();
+    ///                 false
+    ///             }
+    ///         }
+    ///     }
+    ///
+    ///     fn view<'a>(&self, ctx: &'a Context<Self>) -> Html {
+    ///         let onclick = ctx.link().callback_event(|e| RBMessage::Refresh);
+    ///
+    ///         html! {
+    ///             <a href="#" {onclick}>{ "Refresh" }</a>
+    ///         }
+    ///     }
+    /// }
+    /// ```
+    pub fn callback_event<EVT, F>(&self, function: F) -> Callback<EVT>
+    where
+        F: Fn(&EVT) -> COMP::Message + 'static,
+        EVT: Into<web_sys::Event>,
+    {
+        self.callback(move |event: EVT| {
+            let msg = function(&event);
+            let evt: web_sys::Event = event.into();
+            evt.prevent_default();
```

Comment:
I think we should not make a method that calls `prevent_default()` implicitly.

This method is useful in very specific situation (e.g.: suppressing navigation on anchor elements.).

The name `callback_event` encourages users to use it instead of `callback` for event handlers.

Users may create the following implementation:

```rust
fn view(&self, ctx: &Context<Self>) -> Html {
    let oninput = ctx.link().callback_event(|e| ...);

    html! { <input type="checkbox" {onclick} /> }
}
```

And it won't let users check the checkbox because of `preventDefault()`.
This could be difficult to notice.